tasks/src/tasks/authenticate.rs

```diff
@@ -67,27 +92,47 @@ impl Task for AuthenticateTask {
 
     fn process_continuation_request_authenticate(
         &mut self,
-        _: CommandContinuationRequest<'static>,
+        continuation: CommandContinuationRequest<'static>,
     ) -> Result<AuthenticateData<'static>, CommandContinuationRequest<'static>> {
-        if self.ir {
-            Ok(AuthenticateData::Cancel)
-        } else if let Some(line) = self.line.take() {
-            Ok(AuthenticateData::r#continue(line))
+        let data = if self.ir {
```

Comment:
Here, we only ask for an error when SASL-IR was uses. But from my reading of the https://developers.google.com/gmail/imap/xoauth2-protocol, we should also handle the case if SASL-IR was not used in a similar way.